src/otx/algo/object_detection_3d/monodetr3d.py

```diff
@@ -0,0 +1,258 @@
+                biases += [param]
+            else:
+                weights += [param]
+
+        return [{"params": biases, "weight_decay": 0}, {"params": weights, "weight_decay": 0.0001}]
+
+    def forward_for_tracing(
+        self,
+        images: torch.Tensor,
+        calib_matrix: torch.Tensor,
+        img_sizes: torch.Tensor,
+    ) -> dict[str, torch.Tensor]:
+        """Model forward function used for the model tracing during model exportation."""
+        outputs = self.model(images=images, calibs=calib_matrix, img_sizes=img_sizes, mode="predict")
+
+        return {
+            "scores": outputs["pred_logits"].sigmoid(),
+            "boxes_3d": outputs["pred_boxes"],
+            "size_3d": outputs["pred_3d_dim"],
+            "heading_angle": outputs["pred_angle"],
+            "depth": outputs["pred_depth"],  # depth + deviation
+        }
+
+    @staticmethod
+    def extract_dets_from_outputs(outputs: dict[str, torch.Tensor], topk: int = 50) -> tuple[torch.Tensor, ...]:
+        """Extract detection results from model outputs."""
+        # b, q, c
+        out_logits = outputs["pred_logits"]
+        out_bbox = outputs["pred_boxes"]
+
+        prob = out_logits.sigmoid()
+        topk_values, topk_indexes = torch.topk(prob.view(out_logits.shape[0], -1), topk, dim=1)
+
+        # final scores
+        scores = topk_values
+        # final indexes
+        topk_boxes = (topk_indexes // out_logits.shape[2]).unsqueeze(-1)
+        # final labels
+        labels = topk_indexes % out_logits.shape[2]
+
+        heading = outputs["pred_angle"]
+        size_3d = outputs["pred_3d_dim"]
+        depth = outputs["pred_depth"]
+        # decode boxes
+        boxes_3d = torch.gather(out_bbox, 1, topk_boxes.repeat(1, 1, 6))  # b, q', 4
+        # heading angle decoding
+        heading = torch.gather(heading, 1, topk_boxes.repeat(1, 1, 24))
+        # depth decoding
+        depth = torch.gather(depth, 1, topk_boxes.repeat(1, 1, 2))
+        # 3d dims decoding
+        size_3d = torch.gather(size_3d, 1, topk_boxes.repeat(1, 1, 3))
+        # 2d boxes of the corners decoding
+
+        return labels, scores, size_3d, heading, boxes_3d, depth
+
+    @property
+    def _exporter(self) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
+        if self.input_size is None:
+            msg = f"Input size attribute is not set for {self.__class__}"
+            raise ValueError(msg)
+
+        return OTXObjectDetection3DExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=(1, 3, *self.input_size),
+            mean=self.mean,
+            std=self.std,
+            resize_mode="standard",
+            swap_rgb=False,
+            via_onnx=False,
+            onnx_export_configuration={
+                "input_names": ["images", "calib_matrix", "img_sizes"],
+                "output_names": ["scores", "size_3d", "heading_angle", "boxes_3d", "depth"],
```

Comment:
This field is not required: `output_names` are already specified below and would be propagated to ONNX config

Reply:
Done